src/libsyntax/ast.rs

```diff
@@ -19,13 +19,10 @@ use attr::{ThinAttributes, HasAttrs};
 use codemap::{mk_sp, respan, Span, Spanned, DUMMY_SP, ExpnId};
 use abi::Abi;
 use errors;
-use ext::base;
-use ext::tt::macro_parser;
 use parse::token::{self, keywords, InternedString};
-use parse::lexer;
-use parse::lexer::comments::{doc_comment_style, strip_doc_comment_decoration};
 use print::pprust;
 use ptr::P;
+use tokenstream::{TokenTree};
```

Comment:
IMO, no. Token streams will eventually be added to `tokenstream.rs`, and hopefully places that currently use tokens trees will generally move to using token streams. Looking forward, procedural macros will shift to work over token streams, and the breakage here is (at least partially) intended to  ensure that tokentree actions / procedural macros no longer need to rely on `ast.rs` in any way.